vllm/model_executor/models/llama.py

```diff
@@ -352,7 +352,10 @@ def forward(
                                             kv_caches[i - self.start_layer],
                                             attn_metadata, residual)
             if is_hpu and i % self.config_hidden_layers == 0:
-                htorch.core.mark_step()
+                if htorch.utils.internal.is_lazy():
+                    htorch.core.mark_step()
+                else:
+                    torch._dynamo.graph_break()
```

Comment:
In eager mode the test works as expected, so the graph break might be an empty call.
Could you specify the test to be execute so we can use the same for evaluating performance metrics?

Reply:
The test:
              PT_HPU_LAZY_MODE=0  python -u vllm-fork/benchmarks/benchmark_throughput.py \
                        --model /mnt/weka/data/pytorch/llama2/Llama-2-7b-hf \
                        --device hpu \
                        --seed 2024 \
                        --backend vllm \
                        --dataset /software/users/abala/ShareGPT_V3_unfiltered_cleaned_split.json \
                        --num-prompts 1000 \
                        --dtype bfloat16 \
                        --max-model-len 4096 \
                        --max-num-batched-tokens 8192 \
                        --max-num-seqs 128 \
                        --use-padding-aware-scheduling 
But also @bsochack is opposite to use graph_break as rather dubug and internal pytorch API  with other consequences.
BDW, we are now working on adding regional compilation (https://pytorch.org/tutorials/recipes/regional_compilation.html) which also  probably fix your issue ( as I understand is related to very long graph)

Reply:
@anko-intel you are right, we do see performance drop with this change. Since issue we are trying to fix is related to long graph, we would like to understand when we can expect regional compilation to be available, is there a PR for this change which can try out?

Reply:
I hope within 2 weeks we can finish it. We  don't have PR yet. For now you can take for test this commit: https://github.com/Kacper-Pietkun/vllm-fork/commit/1dc2e7ee64313fd27c96160f16f3cac7da6285c2 
You can notice that increasing the cache_size also fix performance in solution with graph breaks, but as I previously have written we would like to avoid using  _dynamo,graph_break().
Probably also increasing cache size 3 times will be enough ( in final version it probably will be not needed at all)

Reply:
just fyi, the target solution should be regional compilation instead of inserting graph_breaks as it can have many unexpected side-effects in torch.compile.
I'm not objecting to merge this change temporally as long as we will replace it with regional compilation and it helps other teams to progress.

Reply:
@vivekgoe : regional compilation https://github.com/HabanaAI/vllm-fork/pull/576 is ready to merge